dspace-api/src/main/java/org/dspace/matomo/MatomoHelper.java

```diff
@@ -0,0 +1,668 @@
+        MATOMO_API_URL_CACHED = configurationService.getProperty("lr.statistics.api.cached.url");
+        AUTH_TOKEN = configurationService.getProperty("lr.statistics.api.auth.token");
+        MATOMO_SITE_ID = configurationService.getProperty("lr.statistics.api.site_id");
+        MATOMO_DOWNLOAD_SITE_ID = configurationService.getProperty("lr.tracker.bitstream.site_id");
+    }
+
+    /**
+     *
+     * @param keys - The order of keys should match the json object indexes
+     * @param report - The downloaded json
+     * @return
+     * @throws Exception
+     */
+    static String transformJSONResults(Set<String> keys, String report) throws Exception {
+        JsonNode json = OBJECT_MAPPER.readTree(report);
+        ObjectNode views = null;
+        ObjectNode downloads = null;
+        int i = 0;
+        for (String key : keys) {
+            if (key.toLowerCase().contains("itemview")) {
+                views = mergeJSONReports(views, (ObjectNode)json.get(i));
+            } else if (key.toLowerCase().contains("downloads")) {
+                downloads = mergeJSONReports(downloads, (ObjectNode)json.get(i));
+            }
+            i++;
+        }
+        ObjectNode response = OBJECT_MAPPER.createObjectNode();
+        ObjectNode result = OBJECT_MAPPER.createObjectNode();
+        result.set("views", (views == null ? OBJECT_MAPPER.createObjectNode() : transformJSON(views)));
+        result.set("downloads", (downloads == null ? OBJECT_MAPPER.createObjectNode() : transformJSON(downloads)));
+        response.set("response", result);
+
+        return response.toString();
+    }
+
+    private static ObjectNode mergeJSONReports(ObjectNode o1, ObjectNode o2) {
+        if (o1 == null) {
+            return o2;
+        } else {
+            //just concatenate the dmy arrays, transformJSON should do the rest
+            Iterator<String> keys = o1.fieldNames();
+            while (keys.hasNext()) {
+                String dmyKey = keys.next();
+                if (o2.get(dmyKey) != null) {
+                    o1.withArray(dmyKey).addAll(o2.withArray(dmyKey));
+                }
+            }
+            keys = o2.fieldNames();
+            while (keys.hasNext()) {
+                String dmyKey = keys.next();
+                if (o1.get(dmyKey) == null) {
+                    o1.set(dmyKey, o2.get(dmyKey));
+                }
+            }
+            return o1;
+        }
+    }
+
+    private static ObjectNode transformJSON(ObjectNode views) {
+        ObjectNode result = OBJECT_MAPPER.createObjectNode();
+        ObjectNode total = OBJECT_MAPPER.createObjectNode();
+        Iterator<String> keys = views.fieldNames();
+        while (keys.hasNext()) {
+            String key = keys.next();
+            ArrayNode view_data = views.withArray(key);
+            if (view_data.isEmpty()) {
+                continue;
+            }
+            String[] dmy = key.split("-");
```

Comment:
Maybe some doc what is dmy = 1; = 2; = n.. it is really hard to understand and review